src/testArrays/ArrayCopy.java

```diff
@@ -0,0 +1,24 @@
+package testArrays;
+
+import java.util.Arrays;
+
+public class ArrayCopy {
+    public static void main(String[] args) {
+        int[] first = {1, 5, 23, 11};
+        int[] second = {3, 12, 7, 8, 10, -4};
+        arraycopy(first, 1, second, 2, 2);
+        System.out.println(Arrays.toString(second));
+    }
+
+    private static void arraycopy(int[] arr1, int sourcePos, int[] arr2, int destPos, int length) {
+        if (arr1 == null || arr2 == null) {
+            throw new NullPointerException("Source or destination array couldn't be empty");
+        }
+        if (sourcePos > arr1.length || destPos > arr2.length) {
```

Comment:
Have you tested this conditions in edge cases? Looks like you allow to copy from position 1 of array with size 1. Which will lead to `IndexOutOfBoundsException` on read or write.